Prog/Births_2009_New.sas

```diff
@@ -0,0 +1,164 @@
+/**************************************************************************
+ Program:  Births_2009_New.sas
+ Library:  Vital
+ Project:  NeighborhoodInfo DC
+ Author:   Rob Pitingolo
+ Created:  6/4/18
+ Version:  SAS 9.4
+ Environment:  Windows with SAS/Connect
+ 
+ Description:  Read-in raw birth data and create births_yyyy
+
+ Modifications:
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( Vital )
+%DCData_lib( MAR )
+
+libname vitalraw "&_dcdata_r_path\Vital\Raw\2018";
+
+%let revisions = New File;
+
+
+** Read raw data and clean some of the variables **;
+data births;
+	set vitalraw.B9final;
+
+	** Convert some character variables to numeric **;
+	mage_n = 1 * mage;
+	bweight_n = 1 * bweight;
+	gest_age_n = 1 * gest_age;
+	num_visit_n = 1 * num_visit;
+
+	** Code missings **;
+	if mage_n = 99 then mage_n = .u;
+	if bweight_n = 9999 then bweight_n = .u;
+	if gest_age_n = 99 then gest_age_n = .u;
+	if num_visit_n = 99 then num_visit_n = .u;
+
+	 ** Check birth dates **;
+  
+	if not missing( date ) and year( date ) ~= 1 * birthyr then do;
+		%warn_put( msg="Birth date in wrong year: " _n_= date= birthyr= birthmo= birthdy= )
+	end;
+
+	** Record race and ethnicity **;
+	if mrace = "White" then mrace_num = 1;
+		else if mrace = "Black" then mrace_num = 2;
+		else if mrace = "Asian/Other" then mrace_num=4;
+		else mrace_num = 0;
+
+	if latino_new = "Non-Hispanic" then latino = "N";
+		else if latino_new = "Unknown" then latino ="N";
+		else if latino_new = "Hispanic" then latino = "Y";
+
+	** Recode marital status **;
+	if mstatnew = "Married" then mstat = 1;
+		else if mstatnew = "Unmarried" then mstat = 2;
+		else mstat = .u;
+
+	** Fill in missing vars for 2010 - 2016 data **;
```

Comment:
There is extensive missing data in 2009 so we should not calculate _any_ of the prenatal care vars.